notes-app-server/src/controllers/v1/notes.handler.ts

```diff
@@ -0,0 +1,103 @@
+import { Request, Response } from "express";
+import { NoteIdSchema, NoteSchema } from "../../validation/note.schema";
+import { NotesService } from "../../services/notes.service";
+
+const getListOfNotes =
+  (service: NotesService) => async (_req: Request, res: Response) => {
+    try {
+      const notes = await service.getNotes();
+
+      return res.send(notes);
+    } catch (error) {
+      console.error(error);
+      return res.status(500).send("Oops! Something went wrong.");
+    }
+  };
+
+const createNote =
+  (service: NotesService) => async (req: Request, res: Response) => {
+    const parseResult = NoteSchema.safeParse(req.body);
+    if (!parseResult.success) {
+      return res
+        .status(400)
+        .send(
+          `Invalid request body.${parseResult.error.errors
+            .map((e) => e.message)
+            .join(", ")}`
+        );
+    }
+
+    try {
+      const note = await service.createNote(parseResult.data);
```

Comment:
Notice, that the handling of "business logic" is forwarded to the service. Only the request & response handling is done on the handler.